Scripts/Generate.ps1

```diff
@@ -88,20 +88,32 @@ namespace Basic.Reference.Assemblies
 "@
 
     $metadataContent += @"
-            public static PortableExecutableReference $propName { get; } = AssemblyMetadata.CreateFromImage(Resources.$($propName)).GetReference(filePath: "$dllName", display: "$dll ($lowerName)");
+            private static PortableExecutableReference? $fieldName;
+            public static PortableExecutableReference $propName
+            {
+                get
+                {
+                    if ($fieldName == null)
+                    {
+                        $fieldName = AssemblyMetadata.CreateFromImage(Resources.$($propName)).GetReference(filePath: "$dllName", display: "$dll ($lowerName)");
+                    }
+                    return $fieldName;
+                }
+            }
+
 
 "@
 
   }
 
   $refInfoContent += @"
-            public static IEnumerable<ReferenceInfo> All { get; }= new []
+            public static IEnumerable<ReferenceInfo> All => new []
             {
 
 "@
 
   $metadataContent += @"
-            public static IEnumerable<PortableExecutableReference> All { get; }= new PortableExecutableReference[]
+            public static IEnumerable<PortableExecutableReference> All => new PortableExecutableReference[]
```

Comment:
Think we need to have a similar delay load experience for the arrays. These are essentially cached today and that is depended on by a few customers. Consider dotnet/roslyn for example accesses the properties a ~100,000 times in a single test run.

Reply:
Sounds good, thanks. 93d28e4